catalogue/models/component/__init__.py

```diff
@@ -202,3 +209,24 @@ def get_technical_specification(self) -> Tuple[str, List[str]]:
 
     def get_source(self) -> Tuple[str, List[str]]:
         return Source._meta.verbose_name, [x.name for x in Source._meta.get_fields()]
+
+@receiver(signals.pre_save, sender=Component)
+def init_unapproved_state(sender, instance, *args, **kwargs):
+    previous = Component.objects.get(id=instance.id)
+    if previous.approved == instance.approved:
+        instance.approved = False
+        context = {
+            "comp": instance,
+        }
+        content = render_to_string("catalogue/emails/email_approve.txt", context)
+        admin_emails = (
+                get_user_model()
+                .objects.filter(groups__name__in=["Moderatoren"])
+                .values_list("email", flat=True)
+            )
+        send_mail(
```

Comment:
Better in post_save or is this fine?

Reply:
I need to check for previous since i need the last state of approved. The email should only be sent if a user changes something, but not if an Admin changes the approved attribute from false to true. Hence i need to check for previous because it has to exists so i can access the approved attribute from it. I couldn't find a way to check in the signal function, which user group did the last modification. Maybe would require a new field in component like "last_modified_by".

Reply:
...the approved field should not go to false if it was changed to true by the admin. For this, i also need the previous state.

Reply:
Maybe a good way would be to add a field in the model, indicating the last modifier.

Reply:
Than i just need to check in the instance if it was modified by a user and set the attribute to false + send email in post save.

Reply:
something like this:
```
if modified by user:
  reset attribute
  if newly created or previously approved:
    send mail
```
django also keeps logs on who changed an object through the admin interface, maybe you can use them instead of adding an additional `last_change_by` attribute.
https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#logentry-objects